### ISSUE TITLE
Some problem of 2nd training processed

### ISSUE THREAD

**Issue by newplay** [QuantumLab-ZY/HamGNN#9]
Dear Yang Zhong,

I have tried the new database, and it seems that the new dataset can improve the mean absolute error (MAE) more effectively. Therefore, I believe the problem may arise from the distance between atoms (in bilayer 2D materials, I suspect it's due to the constraints of van der Waals forces).

Another issue pertains to the readme.md file. You mentioned that we should perform the second training for the band energy to ensure a smaller error:

> Details of training for bands (The 2nd training step)
> When the training of the Hamiltonian matrix is completed in the first step, it is necessary to use the trained network weights > to initialize the HamGNN network and start training for the energy bands. The parameters related to energy band training are as follows:
> 
> The checkpoint_path parameter should be set to the path of the weight file obtained after training on the Hamiltonian matrix in the first step.
> Set load_from_checkpoint to True.
> - The learning rate (lr) should not be too large; it is recommended to use 0.0001.
> - In losses_metrics and metrics, remove the commented section for band_energy.
> - Set calculate_band_energy to True and specify the parameters num_k, band_num, and k_path.

## I have a question: What is the permissible value for the loss in the second training? ##

Additionally, could I request more details about the `twist_bilayer_MoS2` demo? I believe there may be more overlap between this demo and my project. In my project, I am currently aiming to train the `twist_bilayer_$WSe_2$` first. I have used 500 data points for the training process, and I have the graph_data.npz file of my training database below. 

My data is generated using the following procedure:

1. Create the structure from a unit cell monolayer to a $3 \times 3$ supercell.
2. Separate the unit cell into a $10 \times 10$ zone of the a-axis and b-axis.
3. Create the bilayer (2nd layer).
4. Move the 2nd layer step by step as described in step 2.
5. Add some perturbation to each atom after step 4.

**Comment by newplay** [comment]
Thank you for your response. It helped me gain a better understanding of certain aspects. However, I still have some questions regarding the `twist_bilayer_MoS2` demo:

1. How many datas were used for training in the `bilayer_MoS2` demo?
2. How did you generate the data for the perturbation structure? Were perturbations applied along the c-axis?

In my database, I gradually adjusted the interlayer distance to the original interlayer distance minus 0.6 angstroms (equivalent to adding pressure), and I suspect that this alteration may be the reason for the difficulty in achieving convergence during training. However, I am unsure why adding pressure to the structure would cause problems in GNN. That's the reason I want to know in question 2.

Best regards,
TzuChing

**Comment by QuantumLab-ZY** [comment]
Dear TzuChing,

Details of the `bilayer_MoS2` demo can be found in my article:

"Before predicting the electronic structure of the Moiré twisted bilayer MoS2 superlattice, we trained a HamGNN
model using a dataset consisting of 500 untwisted bilayer MoS2 structures, each containing 54 atoms. Each MoS2 bilayer structure in the dataset has a random interlayer sliding distance of up to 2 angstroms along a random direction. The layer spacing of each MoS2 bilayer structure in the dataset was randomly shifted by a maximum of 0.5 angstroms."

Maybe some of the structures in your training set are not reasonable. I suggest that you conduct some checks on the structures in the training set, such as examining the interatomic distance.

Best wishes,
Yang Zhong

**Comment by newplay** [comment]
Dear Yang Zhong:
Thank you for your reply; it seems I have overlooked too many details,I'm sorry about that. I will re-read your paper to ensure I haven't missed any nuances regarding the usage methods.
Best regards,
TzuChing

**Comment by newplay** [comment]
Dear Yang Zhong,

I noticed that the training data you used for MoS2 bilayer appears to be a 3x3 supercell as you mentioned. However, the lattice constant of this supercell is approximately twice as large as that of a normal supercell. Could you please provide some insight into the reason behind this?

Thank you.
ps:
your structure I read from the `MoS2_bilayer_graph_data.npz`:
![image](https://github.com/QuantumLab-ZY/HamGNN/assets/21427346/512f3ca0-8bea-4c1f-b1fc-5bb972317a5b)
![image](https://github.com/QuantumLab-ZY/HamGNN/assets/21427346/0ed5e0da-e886-4e19-b810-720e9aa35a00)
my `WSe2 data`:
![image](https://github.com/QuantumLab-ZY/HamGNN/assets/21427346/9175062e-14bf-40e8-b38f-06594dd75e67)
![image](https://github.com/QuantumLab-ZY/HamGNN/assets/21427346/a3b7166c-48c1-4802-b6fb-8b451e16875f)

**Comment by newplay** [comment]
> Dear Yang Zhong,
> 
> I noticed that the training data you used for MoS2 bilayer appears to be a 3x3 supercell as you mentioned. However, the lattice constant of this supercell is approximately twice as large as that of a normal supercell(a,b axis). Could you please provide some insight into the reason behind this?
> 
> Thank you. ps: your structure I read from the `MoS2_bilayer_graph_data.npz`: ![image](https://private-user-images.githubusercontent.com/21427346/312013327-512f3ca0-8bea-4c1f-b1fc-5bb972317a5b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAyMzYzNjQsIm5iZiI6MTcxMDIzNjA2NCwicGF0aCI6Ii8yMTQyNzM0Ni8zMTIwMTMzMjctNTEyZjNjYTAtOGJlYS00YzFmLWIxZmMtNWJiOTcyMzE3YTViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEyVDA5MzQyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkNTM0Yjk0ZmFkODYwMDFkMjNmYjYyY2ZjNGI1Y2EwODRiZmQ0NDRiZjBhMGU4NTRhZTU5ZGEzYzRlNGNjMWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.E6KIjE2iPf8dyxizAnG4GpDz_Vv6eG_7CI16SYDrPeM) ![image](https://private-user-images.githubusercontent.com/21427346/312014273-0ed5e0da-e886-4e19-b810-720e9aa35a00.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAyMzYzNjQsIm5iZiI6MTcxMDIzNjA2NCwicGF0aCI6Ii8yMTQyNzM0Ni8zMTIwMTQyNzMtMGVkNWUwZGEtZTg4Ni00ZTE5LWI4MTAtNzIwZTlhYTM1YTAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEyVDA5MzQyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyN2FmYjczYzdlZjZmZTlhYTdmMGVjNjYwZDUwMzRiNmY1YWU4YjI1ZDgwNjU5ZjUzNGE1MzZlOTdiYjQ2ZDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.963RHRdcMI-CVz-uUDiIVBcmTgGkXRRt5Ie1weg1Ur0) my `WSe2 data`: ![image](https://private-user-images.githubusercontent.com/21427346/312014703-9175062e-14bf-40e8-b38f-06594dd75e67.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAyMzYzNjQsIm5iZiI6MTcxMDIzNjA2NCwicGF0aCI6Ii8yMTQyNzM0Ni8zMTIwMTQ3MDMtOTE3NTA2MmUtMTRiZi00MGU4LWIzOGYtMDY1OTRkZDc1ZTY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEyVDA5MzQyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3NmIyZGIxZGZjMjYyNGEyZjk0YjQ1ZDM3YzQ1NDJiYWNlYmQyMWU4N2VjZTI5NGRmOTM4ZjM1NTk1MmJkY2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.nF3SFIf8vX1DOYWp_5BmnM8GtkhtLYUEl_3SWph6k6c) ![image](https://private-user-images.githubusercontent.com/21427346/312014787-a3b7166c-48c1-4802-b6fb-8b451e16875f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAyMzYzNjQsIm5iZiI6MTcxMDIzNjA2NCwicGF0aCI6Ii8yMTQyNzM0Ni8zMTIwMTQ3ODctYTNiNzE2NmMtNDhjMS00ODAyLWI2ZmItOGI0NTFlMTY4NzVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEyVDA5MzQyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYwMWVhZGJjMWY0YjhmZDM2ZTVjMzQ2ZjVkZjQyNTFhMmZlNGIzNjk2ZmJlZGQ4NjA3NjU1NDcyN2NmMzNmOWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3wYDVOzoBwx0h_pI4f_qOc72bjmkk_nB-cBLVaT0gBw)

sorry , I find that the unit is `au` , not `angstrom` .
